### DISCUSSION TITLE
对于雷电模拟器端口变动导致连接失败的解决办法

### DISCUSSION THREAD

**Discussion by gClov** [MaaAssistantArknights/MaaAssistantArknights#2930] [Show and tell]
### 在提问之前...

- [X] 我理解 Issue 是用于反馈和解决问题的，而非吐槽评论区，且会尽可能提供更多信息帮助问题解决
- [X] 我在提问题之前至少花费了 5 分钟来思考和准备
- [X] 我已经阅读了文档中的常见问题、详细使用说明等，认为这大概率是一个 bug 而非设计如此
- [X] 我已查看最新测试版本的更新内容，并未提及该 bug 已被修复的情况
- [X] 我上传的游戏画面截图是模拟器自带的截图工具（或通过 adb）截取的原图

### 问题描述

使用adb命令 adb devices 查看连接地址

详细步骤为在雷电模拟器adb目录下打开cmd 
![image](https://user-images.githubusercontent.com/115129135/204171040-1ef1042a-ccd8-4fc3-9edf-b2ca8349e24a.png)

在cmd中输入adb devices（如后面显示offline，则在任务管理器—详情信息—结束名为adb的进程后再次使用命令查看即可）
![image](https://user-images.githubusercontent.com/115129135/204170970-344cbc79-2b33-4712-81f5-9028fbab96fd.png)

找到了有明日方舟模拟器连接地址后直接在maa连接设置修改就好。（基本命名规则如下）

maa连接失败的原因是默认连接地址是127.0.0.1:5555（或emulator-5554
但多开后第（序号）个模拟器端口号为5555+序号*2（5554+序号*2
和
雷电模拟器有两套不同的（连接地址命名规则？）
可能你一开始是127.0.0.1:5555下次打开就是emulator-5554了
![image](https://user-images.githubusercontent.com/115129135/204171347-b174513b-b93e-48d1-b0f7-551154bfcd32.png)

找到了有明日方舟模拟器连接地址后直接在maa连接设置修改就好。

（那个日志应该没什么用，不提交传不上来）
（我知道蓝叠国际版已经是调好的了，但是它本身的bug真的用起来不舒服，而且他的多开真的比雷电卡，看内存也没爆，突然电脑就卡的一顿一顿了，还以为是之前搞虚拟机把显卡驱动弄坏了，虚拟机也删了，驱动也重装了www）

### 日志文件

[asst.log](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/10099750/asst.log)

**Comment by ntgmc** [comment]
额，没看懂，发这个issue干嘛

**Comment by gClov** [comment]
因为maa在雷电模拟器使用时经常因为这个导致没法使用，官方文档上也没解决办法。
可这个改下就几乎就没什么问题了，感觉这个可以放到文档或者常见问题里

**Comment by ABA2396** [comment]
> 因为maa在雷电模拟器使用时经常因为这个导致没法使用，官方文档上也没解决办法。 可这个改下就几乎就没什么问题了，感觉这个可以放到文档或者常见问题里

连接检测里有一个选项是一直检测，打开就行了

**Reply by xiaoweiXX** [reply]
> > 因为maa在雷电模拟器使用时经常因为这个导致没法使用，官方文档上也没解决办法。 可这个改下就几乎就没什么问题了，感觉这个可以放到文档或者常见问题里
> 
> 连接检测里有一个选项是一直检测，打开就行了

如果打开这个一直检测，那真的会每次MAA点Link Start！（也就是每次开始MAA任务）就会检测一遍IP和端口，会浪费很多时间。之前我提议每次打开MAA时检测一遍好像没被采纳来着。

**Reply by ABA2396** [reply]
> 如果打开这个一直检测，那真的会每次MAA点Link Start！（也就是每次开始MAA任务）就会检测一遍IP和端口，会浪费很多时间。之前我提议每次打开MAA时检测一遍好像没被采纳来着。

雷电很早之前就改过了，现在固定的就是emulator-5554

**Reply by xiaoweiXX** [reply]
> > 如果打开这个一直检测，那真的会每次MAA点Link Start！（也就是每次开始MAA任务）就会检测一遍IP和端口，会浪费很多时间。之前我提议每次打开MAA时检测一遍好像没被采纳来着。
> 
> 雷电很早之前就改过了，现在固定的就是emulator-5554

那就好。我现在已经养成习惯了，每次点开MAA先点开检测更新，然后跑一遍登录。

**Comment by ABA2396** [comment]
![image](https://user-images.githubusercontent.com/99072975/204200864-e357a7fe-94d3-4d8d-960c-2118ad653656.png)

**Comment by gClov** [comment]
这个在我这里没成功过（笑哭
我的是雷电模拟器，应该没做兼容什么的

**Comment by ABA2396** [comment]
> 这个在我这里没成功过（笑哭 我的是雷电模拟器，应该没做兼容什么的

在和雷电对接，之后他们可能会改成只用127.0.0.1:5555这种格式

**Comment by MistEO** [comment]
我们和雷电的开发同学沟通了，会给我们一个 IP 不会变动的版本，在测试没问题后他们会更新雷电自带的 adb

**Reply by MistEO** [reply]
另外这个放在讨论区感觉好一点，我直接转过来了

**Reply by gClov** [reply]
好的，感谢
